exe/wallet/Main.hs

```diff
@@ -385,9 +389,11 @@ execHttpBridge args _ = do
     waitForConnection nw defaultRetryPolicy
     let tl = HttpBridge.newTransactionLayer @n
     wallet <- newWalletLayer @_ @(HttpBridge n) tracer block0 db nw tl
-    let logStartup port = TIO.hPutStrLn stderr $
+    let logStartup port = logInfo tracer $
             "Wallet backend server listening on: " <> toText port
-    Server.start logStartup walletPort wallet
+    Server.withListeningSocket walletPort $ \(port, socket) -> do
+        let settings = Server.mkWarpSettings logStartup port
+        race_ (daedalusIPC port) (Server.startOnSocket settings socket wallet)
```

Comment:
Ah! I supposed this is why we make the `daedalusIPC` hangs forever. I believe that switching for `concurrently` instead of `race_` would remove that limitation since the action could return without cancelling the `startOnSocket`.

Reply:
This needs to be `race` so that the program exits when there is an error from `daedalusIPC`.

Reply:
Got you, [this comment](https://github.com/input-output-hk/cardano-wallet/pull/388#discussion_r293646245) of yours clarifies the choice of `race` indeed. I'd suggest to have the behavior documented either on the call-site or, somewhere in the DaedalusIPC. Maybe in both places.